src/ShaderSwapper.Core/ShaderSwapper.cs

```diff
@@ -30,62 +30,190 @@ public class ShaderSwapper : BaseUnityPlugin
         public const string PluginGUID = "com.deathweasel.bepinex.shaderswapper";
         public const string PluginName = "Shader Swapper";
         public const string PluginNameInternal = Constants.Prefix + "_ShaderSwapper";
-        public const string PluginVersion = "1.6";
+        public const string PluginVersion = "1.7";
         internal static new ManualLogSource Logger;
         private static ShaderSwapper Instance;
 
+        internal class SwapTargetList
+        {
+            internal class Rule
+            {
+                internal class EntryList
+                {
+                    public static string EntryElementName { get; } = "Entry";
```

Comment:
How about naming it "Shader_Name" or similar instead in order to make it easier to understand on a glance? It would also allow for different elements like layer or GameObject name/path in the future. It could even be done with reflection I guess, e.g. "Material_xyz" which checks property "xyz" on the Material.

Reply:
I think leaving the element name as "Entry" would be better, and instead changing the surrounding element name (i.e. "Include", "Exclude") to describe different behaviours and rules? Can definitely foresee this feature being expanded in the future. Thoughts?

Reply:
Whichever way works for you, really.